googletest/test/gtest_unittest.cc

```diff
@@ -3103,6 +3103,103 @@ TEST_F(DoubleTest, DoubleLEFails) {
 }
 
 
+TEST(LongDoubleTest, EXPECT_NEAR) {
+  {
+    long double a = -1.0;
+    long double b = -1.1;
+    EXPECT_NEAR(a, b, 0.2);
+  }
+  {
+    long double a = -1.0;
+    long double b = -1.5;
+    EXPECT_NONFATAL_FAILURE(EXPECT_NEAR(a, b, 0.25),  // NOLINT
+                            "The difference between a and b is 0.5, "
+                            "which exceeds 0.25, where\na evaluates to -1,\nb "
+                            "evaluates to -1.5, and\n0.25 evaluates to 0.25.");
+  }
+}
+
+TEST(LongDoubleTest, ASSERT_NEAR) {
+  {
+    long double a = -1.0;
+    long double b = -1.1;
+    ASSERT_NEAR(a, b, 0.2);
+  }
+  {
+    EXPECT_FATAL_FAILURE(long double a = -1.0; long double b = -1.5;
+                         ASSERT_NEAR(a, b, 0.25),  // NOLINT
+                         "The difference between a and b is 0.5, "
+                         "which exceeds 0.25, where\na evaluates to -1,\nb "
+                         "evaluates to -1.5, and\n0.25 evaluates to 0.25.");
+  }
+}
+
+namespace {
+// a custom floating-point like type which implements the minimal interface
```

Comment:
After internal discussion, we prefer the same code to look a bit different, please see below:

```suggestion
// a custom floating-point like type which implements the minimal interface
namespace custom_real_ns {

// a custom floating-point like type which implements the minimal interface
// required to test EXPECT_NEAR on a custom type
class custom_real {
  // internally stores as a double to prevent having to implement an actual
  // custom real-like type
  double value_;

 public:
  explicit custom_real(double value) : value_(value) {}

  friend custom_real operator-(custom_real l, custom_real r) {
    return custom_real(l.value_ - r.value_);
  }

  friend bool operator<(custom_real l, custom_real r) {
    return l.value_ < r.value_;
  }
  friend bool operator>(custom_real l, custom_real r) {
    return l.value_ > r.value_;
  }
  friend bool operator<=(custom_real l, custom_real r) {
    return l.value_ <= r.value_;
  }
  friend bool operator>=(custom_real l, custom_real r) {
    return l.value_ >= r.value_;
  }
  friend bool operator==(custom_real l, custom_real r) {
    return l.value_ == r.value_;
  }
  friend bool operator!=(custom_real l, custom_real r) {
    return l.value_ != r.value_;
  }
  friend std::ostream& operator<<(std::ostream& out, custom_real v) {
    out << v.value_;
    return out;
  }
};

}  // namespace custom_real_ns

```